src/Files.App/Data/Factories/ShellContextFlyoutHelper.cs

```diff
@@ -359,20 +359,27 @@ async Task InvokeShellMenuItemAsync(ContextMenu contextMenu, object? tag)
 					itemContextMenuFlyout.SecondaryCommands.Insert(1, sendToItems.FirstOrDefault());
 				}
 
-				// Add items to shell submenu
-				shellMenuItems.Where(x => x.LoadSubMenuAction is not null).ForEach(async x =>
-				{
-					await x.LoadSubMenuAction();
 
+				var itemsWithSubMenu = shellMenuItems.Where(x => x.LoadSubMenuAction is not null).ToList();
+				var subMenuTasks = itemsWithSubMenu.Select(async item =>
+				{
+					await item.LoadSubMenuAction();
 					if (!UserSettingsService.GeneralSettingsService.MoveShellExtensionsToSubMenu)
 					{
-						AddItemsToMainMenu(itemContextMenuFlyout.SecondaryCommands, x);
+						AddItemsToMainMenu(itemContextMenuFlyout.SecondaryCommands, item);
 					}
-					else if (itemContextMenuFlyout.SecondaryCommands.FirstOrDefault(x => x is AppBarButton appBarButton && (appBarButton.Tag as string) == "ItemOverflow") is AppBarButton overflowItem)
+					else
 					{
-						AddItemsToOverflowMenu(overflowItem, x);
+						var overflowItem = itemContextMenuFlyout.SecondaryCommands.FirstOrDefault(x => x is AppBarButton appBarButton && (appBarButton.Tag as string) == "ItemOverflow") as AppBarButton;
+
+						if (overflowItem != null)
+						{
+							AddItemsToOverflowMenu(overflowItem, item);
+						}
 					}
```

Comment:
I don't think this change is necessary.

Reply:
It's the same thing, the old line is too long to read at the IDE

Reply:
Then you should divide new L373.